src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineReceiverTest.php

```diff
@@ -68,6 +71,23 @@ public function testItRejectTheMessageIfThereIsAMessageDecodingFailedException()
         $receiver->get();
     }
 
+    public function testOccursRetryableExceptionFromConnection()
+    {
+        $serializer = $this->createSerializer();
+        $connection = $this->createMock(Connection::class);
+        $driverException = new PDOException(new \PDOException('Deadlock', 40001));
+        $connection->method('get')->willThrowException(new DeadlockException('Deadlock', $driverException));
+        $receiver = new DoctrineReceiver($connection, $serializer);
+        $this->assertSame([], $receiver->get());
+        $this->assertSame([], $receiver->get());
+        $this->expectException(TransportException::class);
+        $receiver->get();
+        $this->assertSame([], $receiver->get());
+        $this->assertSame([], $receiver->get());
+        $this->expectException(TransportException::class);
```

Comment:
this test does not work: once the first exception is thrown, the execution of the test stops. so this part is never executed.

Reply:
ou.. I didn't realise it :) sorry.. it's fixed.